src/include/wiredtiger.in

```diff
@@ -39,6 +39,16 @@ extern "C" {
 #define	__F(func) (*(func))
 #endif
 
+/*
+ * We support configuring WiredTiger with the gcc/clang -fvisibility=hidden
+ * flags, but that requires public APIs be specifically marked.
+ */
+#if defined(DOXYGEN) || defined(SWIG) || !defined(__GNUC__)
+#define	__LIBRARY_VISIBLE
+#else
+#define	__LIBRARY_VISIBLE	__attribute__ ((visibility("default")))
+#endif
+
```

Comment:
> Why not use WT_LIBRARY_VISIBLE here (and not need the internal versions in gcc.h, lint.h and msvc.h) instead of another macro starting with underscores?

I liked the naming better when all of the internal library gcc/clang`#define` attributes were in the same place instead of being split across two files -- I thought it would make more sense when we add support for other attributes in the future. I don't feel strongly, though, I've pushed that change in 94bad5a.